pkg/kubelet/cm/pod_container_manager_linux_test.go

```diff
@@ -293,3 +294,85 @@ func TestGetPodContainerName(t *testing.T) {
 		})
 	}
 }
+
+func Test_podContainerManagerImpl_EnsureExists(t *testing.T) {
+	newGuaranteedPodWithUID := func(uid types.UID) *v1.Pod {
+		return &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				UID: uid,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name: "container",
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								v1.ResourceCPU:    resource.MustParse("1000m"),
+								v1.ResourceMemory: resource.MustParse("1G"),
+							},
+							Limits: v1.ResourceList{
+								v1.ResourceCPU:    resource.MustParse("1000m"),
+								v1.ResourceMemory: resource.MustParse("1G"),
+							},
+						},
+					},
+				},
+			},
+		}
+	}
+
+	type fields struct {
+		cgroupManager CgroupManager
+	}
+	type args struct {
+		pod *v1.Pod
+	}
+	tests := []struct {
+		name            string
+		fields          fields
+		args            args
+		wantErr         bool
+		wantCgroupCalls []string
+	}{
+		{
+			name: "cgroup exists",
+			fields: fields{
+				cgroupManager: &FakeCgroupManager{exists: true},
+			},
+			args:            args{pod: newGuaranteedPodWithUID("fake-uid-1")},
+			wantErr:         false,
+			wantCgroupCalls: []string{"Exists"},
```

Comment:
Also want that nothing else was called. Otherwise test doesn't test much

Reply:
I can make it explicit all calls to the CgroupManager, but I think that might be getting into some of the implementation details.  For example ALL calls to this function will have make a call to `Name()`.  

I could also do a test that some calls aren't made. For example in the first test we want to see calls to `Exist()` and no calls to `Create()`

Is there a preferred approach?

Reply:
ok, test that Create was not called

Reply:
Addressed in https://github.com/kubernetes/kubernetes/pull/127819/commits/09cb1da6954dcace111f8c75d32aadad7a64014f